rust/agama-server/src/web/http.rs

```diff
@@ -69,6 +72,33 @@ pub async fn login(
     Ok((headers, content))
 }
 
+#[derive(Clone, Deserialize, utoipa::ToSchema)]
+pub struct LoginFromQueryParams {
+    /// Token to use for authentication.
+    token: String,
+}
+
+#[utoipa::path(get, path = "/login", responses(
+    (status = 301, description = "Injects the authentication cookie if correct and redirects to the web UI")
+))]
+pub async fn login_from_query(
+    State(state): State<ServiceState>,
+    Query(params): Query<LoginFromQueryParams>,
+) -> impl IntoResponse {
+    let mut headers = HeaderMap::new();
+
+    if TokenClaims::from_token(&params.token, &state.config.jwt_secret).is_ok() {
+        let cookie = format!("agamaToken={}; HttpOnly", params.token);
```

Comment:
There is no expiration set, how long is the cookie valid by default?

Reply:
Not having an expiration date makes it a "session cookie". It should be gone once you close the browser. You can find more information in the [Set-Cookie](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#expiresdate) docs.

Reply:
Ah, when there is no expiration then the cookie should be valid for the current session (until the browser is closed). That's IMHO a correct behavior in our use case.

Reply:
Probably worth adding a comment into the code...